window-management-api/index.js

```diff
@@ -0,0 +1,190 @@
+// Data for the sites to open
+const sites = [
+  {
+    "name": "MDN code playground",
+    "url": "https://developer.mozilla.org/en-US/play"
+  },
+  {
+    "name": "MDN HTML reference",
+    "url": "https://developer.mozilla.org/en-US/docs/Web/HTML"
+  },
+  {
+    "name": "MDN CSS reference",
+    "url": "https://developer.mozilla.org/en-US/docs/Web/CSS"
+  },
+  {
+    "name": "MDN JavaScript reference",
+    "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript"
+  }
+]
+
+const outputElem = document.querySelector("section");
+
+// Array to hold references to the currently open windows
+let windowRefs = [];
+
+// Constants to represent the width and height of the Chrome UI when calculating the window size to open 
+const WINDOW_CHROME_Y = 51;
+const WINDOW_CHROME_X = 1;
+
+// Feature detection: we'll open windows in sites that support the API
+// and create links in sites that do not
+if ("getScreenDetails" in window) {
+  // The Window Management API is supported
+  createButton();
+} else {
+  // The Window Management API is not supported
+  createLinks(sites);
+}
+
+// Functions for creating the links or the button
+
+function createLinks(sites) {
+  const list = document.createElement("ul");
+
+  sites.forEach(site => {
+    const listItem = document.createElement("li");
+    const link = document.createElement("a");
+
+    link.textContent = site.name;
+    link.href = site.url;
+
+    listItem.appendChild(link);
+    list.appendChild(listItem);
+  });
+
+  outputElem.appendChild(list);
+}
+
+function createButton() {
+  const btn = document.createElement("button");
+  btn.textContent = "Open learning environment"
+  outputElem.appendChild(btn);
+
+  btn.addEventListener("click", openWindows);
+}
+
+// Functions for creating the windows
+
+function openWindow(left, top, width, height, url) {
```

Comment:
I'm wondering if it wouldn't be better to have a signature like `openWindow(url, { left, top, width, height })`, i.e. an option object to replace the number parameters?

Reply:
That's fine. Thanks Chris!